### PR TITLE
Streamline FitResult and improve the loss computation

### PR DESCRIPTION
**Pull request by ahwillia**
@erichson - please review and let me know if this looks good to you! If so, feel free to merge.

### CODE REVIEW

File: notes/als_loss.tex [hunk 1]
```diff
@@ -0,0 +1,79 @@
+\documentclass{article}
```
[the remark this answers is not in the document]
Comment (ahwillia) [reply notes/als_loss.tex]: 👍

File: tensortools/optimize/optimize.py [hunk 2]
```diff
@@ -91,47 +50,37 @@ def update2(self, obj):
...
         self.converged =\
-            ( self.iterations > self.min_iter and fit_improvement < self.tol ) or\
+            ( self.iterations > self.min_iter and improvement < self.tol ) or\
             ( self.iterations > self.max_iter or self.time_elapsed() > self.max_time )
 
         return self    
-    
-    
-
-    def compute_fit(self, X):
```
[the remark this answers is not in the document]
Comment (ahwillia) [reply tensortools/optimize/optimize.py]: We can add it back. I think we may need a different `FitResult` to handle randomized algorithms (to handle convergence checking)

File: tensortools/optimize/ncp_als.py [hunk 3]
```diff
@@ -185,18 +185,18 @@ def ncp_als(X, rank=None, random_state=None, **options):
         
         # Compute objective function
         #grams *= U[X.ndim - 1].T.dot(U[X.ndim - 1])
-        #obj = np.sqrt(sci.sum(grams) - 2 * sci.sum(U[X.ndim - 1] * p) + normX**2) / normX
-        obj = sci.linalg.norm(X - U.full()) / normX
+        obj = np.sqrt(sci.sum(grams) - 2 * sci.sum(U[X.ndim - 1] * p) + normX**2) / normX
```
[the remark this answers is not in the document]
Comment (ahwillia) [reply tensortools/optimize/ncp_als.py]: Yes, let's remove `ncp_als` the nonnegative least-sqaures solver I have is not much slower anyways. So I don't think there is an advantage.

File: tensortools/optimize/optimize.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (erichson) [reply tensortools/optimize/optimize.py]: Okay, I like that idea!

**Pull request merged**